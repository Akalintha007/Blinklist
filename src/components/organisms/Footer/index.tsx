import {Container, Box} from '@mui/material';
import Logo from '../../molecules/Logo';
import Typography from '../../atoms/Typography';
const FooterComponent = (props:any)=>{
    const data = [
        {
            'id':601,
            'heading': 'Editorial',
             'topics': [
                 
                 'Book lists',
                 'What is Notfication?',
                 'What to read next?',
                 'Benefits of reading'
             ]
        },
        {
            'id':602,
            'heading': 'Useful links',
             'topics': [
                 'Pricing',
                 'Blinkist business',
                 'Gift cards',
                 'Blinkist magazine'
             ]
        },
        {
            'id':603,
            'heading': 'Company',
             'topics': [
                 'About',
                 'Carrer',
                 'partners',
                 'Code of Conduct'
             ]
        } 
    ]
    return (
        <Box 
        sx={{
          height:'270px',
          width:'100%', 
          backgroundColor: '#F1F6F4',
          display: 'flex',
          position: 'sticky',
          top: '100vh',
          padding: '24px 0px',
          marginTop: '24px'}} >
          <Container sx={{display: 'flex', justifyContent:'start'}} >
            <Box sx={{marginRight:'150px'}}>
                <Logo height= {32} width={32} url= 'https://i.ibb.co/KVbz1Y1/blinklist-logo.png'  data-testid='logo'/>
                <Typography variant="subtitle1" sx={{ color: '#0365F2'}} style={{ textAlign: "left"}} data-testid='slogan'>
                    Big ideas in small packages<br/>
                    Start learning now
                </Typography>
            </Box>
            <Box sx={{display: 'flex', justifyContent: 'end',}}>
                {
                    
                    data.map(curr => {
                        return (<Box key={curr.id} sx={{
                            margin: '0px 32px'

                        }} data-testid='topics'>
                            <Typography variant="body1" pb={2} sx={{display:'flex',justifyContent:'start',color:'#03314B'}}>
                                {curr.heading}
                            </Typography>
                            {
                                curr.topics.map(currTopic => {
                                    return <Typography variant="body2" pb={1} sx={{display:'flex',justifyContent:'start',color:'#6D787E'}} key={currTopic}>
                                        {currTopic}
                                    </Typography>
                                })
                            }
                        </Box>)
                    })
                }
            </Box>
        </Container>
        </Box>   
    )
}

export default FooterComponent;