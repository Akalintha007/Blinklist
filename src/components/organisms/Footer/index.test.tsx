import Footer from '.';
import { render, screen } from '@testing-library/react';


test('Footer', () => {
    render(
        <Footer />
    );
  
  
});

