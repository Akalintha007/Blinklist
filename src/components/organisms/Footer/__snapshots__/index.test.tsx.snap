// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer 1`] = `
<div
  className="MuiBox-root css-1sp4l3i"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1imjkd-MuiContainer-root"
  >
    <div
      className="MuiBox-root css-315n33"
    >
      <div
        className="MuiBox-root css-70qvj9"
      >
        <img
          alt="not found"
          className="MuiBox-root css-xgff9b"
          src="https://i.ibb.co/KVbz1Y1/blinklist-logo.png"
        />
      </div>
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 css-1yfxpjk-MuiTypography-root"
        data-testid="slogan"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        Big ideas in small packages
        <br />
        Start learning now
      </h6>
    </div>
    <div
      className="MuiBox-root css-ckyhk3"
    >
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1u48fnv-MuiTypography-root"
        >
          Editorial
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          Book lists
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          What is Notfication?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          What to read next?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          Benefits of reading
        </p>
      </div>
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1u48fnv-MuiTypography-root"
        >
          Useful links
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          Pricing
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          Blinkist business
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          Gift cards
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          Blinkist magaine
        </p>
      </div>
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1u48fnv-MuiTypography-root"
        >
          Company
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          About
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          Carrer
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          partners
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-kdux84-MuiTypography-root"
        >
          Code of Conduct
        </p>
      </div>
    </div>
  </div>
</div>
`;
